kotlinlib/src/mill/kotlinlib/js/KotlinJSModule.scala

```diff
@@ -0,0 +1,509 @@
+package mill.kotlinlib.js
+
+import mainargs.arg
+import mill.api.{PathRef, Result}
+import mill.define.{Command, Segment, Task}
+import mill.kotlinlib.worker.api.{KotlinWorker, KotlinWorkerTarget}
+import mill.kotlinlib.{Dep, DepSyntax, KotlinModule}
+import mill.scalalib.Lib
+import mill.scalalib.api.CompilationResult
+import mill.testrunner.TestResult
+import mill.util.Jvm
+import mill.{Agg, Args, T}
+import upickle.default.{macroRW, ReadWriter => RW}
+
+import java.io.File
+import java.util.zip.ZipFile
+
+/**
+ * This module is very experimental. Don't use it, it is still under the development, APIs can change.
```

Comment:
Is there any API reference or example code you used to find all the different configuration keys that KotlinJS exposes, and you had to translate to Mill here? If so we should put it in the scaladoc so future readers can see where all this stuff came from

Reply:
That is where I would avoid adding any documentation for now, because this Kotlin/JS module is not ready for being used at all, so it shouldn't be in the docs/example. The first blocker is inability to use `.klib` instead of `.jar`, the second one is API: it is subject to change anyway, and I took an inspiration from `ScalaJSModule`, but API there looks strange - some methods have `scalajs` prefix, some don't, etc.

The point of this PR is just to add the first building block to the Kotlin/JS support.